### PR TITLE
Don't trigger Observers and rewrite PathEntity.tick() for better performance

### PR DESCRIPTION
**Pull request by Fourmisain**
As suspected in #32, the neighbor updates were indeed what triggered the Observers.
I missed that the `PathEntity` also changes the block state each tick, which took a while to figure out.

The fix for #32 was simply using
`world.setBlockState(pos, ..., Block.NOTIFY_LISTENERS | Block.FORCE_STATE);`
everywhere in `PathBlock` and `PathEntity`.

The default flags are `NOTIFY_LISTENERS` and `NOTIFY_NEIGHBORS`, so the above turns off the neighbor updates while `FORCE_STATE` turns off the "new style" neighbor updates - no idea what exactly that means, but we surely don't need or want them.
I also set the `SKIP_LIGHTING_UPDATES` flag as well, since the lighting won't change anyway.

While looking at `PathEntity.tick()` I noticed a lot of room for performance improvement because it calls `world.getBlockState()` and `world.setBlockState()` tons of times when there's no need to.

Since path can be everywhere and is constantly ticking, I took the time to rewrite it - it should be functionally identical the the previous version (except that I reordered it to "upgrade" before "downgrading" in the same tick).

Basically, instead of `world.getBlockState()`, the method parameter `state` is used and changed on the way along until it reaches the (singular) `world.setBlockState()` at the end.
A few other values are cached as well and I changed the `if` structure to not retest already known stuff.

### CODE REVIEW

**Comment by lebonq** [comment]
Thanks you so much ! Good work !
